static_src/actions/user_actions.js

```diff
@@ -156,44 +149,46 @@ const userActions = {
     return Promise.resolve(err);
   },
 
-  receiveUserForCF(user, inviteData) {
-    AppDispatcher.handleServerAction({
-      type: userActionTypes.USER_IN_CF_CREATED,
-      user
+  createUserAndAssociate(data) {
+    const orgGuid = OrgStore.currentOrgGuid;
+    const userGuid = data.userGuid;
+    AppDispatcher.handleViewAction({
+      type: userActionTypes.USER_ORG_ASSOCIATE,
+      userGuid,
+      orgGuid
     });
-
-    if (user.guid) {
-      userActions.sendUserInviteEmail(inviteData);
-    }
-    // Once the user exists in CF, associate them to the organization.
-    return userActions.associateUserToOrg(user);
+    return cfApi.putAssociateUserToOrganization(userGuid, orgGuid)
+      .then(() => cfApi.fetchOrgUsers(orgGuid))
+      .then(orgUsers => userActions.createdUserAndAssociated(userGuid, orgGuid, orgUsers));
   },
 
-  sendUserInviteEmail(inviteData) {
-    AppDispatcher.handleServerAction({
-      type: userActionTypes.USER_EMAIL_INVITE
+  createdUserAndAssociated(userGuid, orgGuid, orgUsers) {
+    AppDispatcher.handleViewAction({
+      type: userActionTypes.USER_ORG_ASSOCIATED,
+      userGuid,
+      orgGuid,
+      orgUsers
     });
-    uaaApi.sendInviteEmail(inviteData);
+    return userActions.createdUserDisplayed(userGuid, orgUsers, orgGuid);
   },
 
-  associateUserToOrg(user) {
-    AppDispatcher.handleServerAction({
-      type: userActionTypes.USER_ORG_ASSOCIATE
+  createdUserDisplayed(userGuid, orgUsers, orgGuid) {
```

Comment:
I think we can use the above `createdUserAndAssociated` action in place of this one.